microprofile.ls/org.eclipse.lsp4mp.ls/src/main/java/org/eclipse/lsp4mp/ls/JavaTextDocumentSnippetRegistry.java

```diff
@@ -67,4 +80,71 @@ public JavaTextDocumentSnippetRegistry() {
 		}
 		return types;
 	}
+
+	@Override
+	public void registerSnippet(Snippet snippet) {
+		preprocessSnippetBody(snippet);
+		super.registerSnippet(snippet);
+	}
+
+	/**
+	 * Preprocess Snippet body for maning package name.
+	 * 
+	 * 
+	 * @param snippet
+	 */
+	private void preprocessSnippetBody(Snippet snippet) {
+		List<String> body = snippet.getBody();
+		if (body.isEmpty()) {
+			return;
+		}
+		String firstLine = body.get(0);
+		if (firstLine.contains("${") && firstLine.contains(PACKAGENAME_KEY)) {
+			// Transform this 3 body lines:
+			// "package ${1:packagename};",
+			// "",
+			// "import org.eclipse.microprofile.health.HealthCheck;",
+
+			// to one line:
+			// ${packagename}import org.eclipse.microprofile.health.HealthCheck;
+
+			if (body.size() >= 2 && StringUtils.isEmpty(body.get(1))) {
+				// Remove the line ""
+				body.remove(1);
+			}
+			String line = "";
+			if (body.size() >= 2) {
+				// Remove the line "import org.eclipse.microprofile.health.HealthCheck;"
+				line = body.get(1);
+				body.remove(1);
+			}
+			// Update the line 0 to ${packagename}import
+			// org.eclipse.microprofile.health.HealthCheck;
+			body.set(0, "${" + PACKAGENAME_KEY + "}" + line);
+		}
+	}
+
+	public List<CompletionItem> getCompletionItems(JavaTextDocument document, int completionOffset,
+			boolean canSupportMarkdown, boolean snippetsSupported,
+			BiPredicate<ISnippetContext<?>, Map<String, String>> contextFilter) {
+		Map<String, String> model = new HashMap<>();
+		// fill package name to replace in the snippets
+		String packageStatement = "";
+		if (!StringUtils.isEmpty(document.getPackageName())) {
+			String lineDelimiter = System.lineSeparator();
+			try {
+				lineDelimiter = document.lineDelimiter(0);
+			} catch (BadLocationException e) {
+			}
+			packageStatement = new StringBuilder("package ")//
+					.append(document.getPackageName()) //
+					.append(";") //
+					.append(lineDelimiter) //
+					.append(lineDelimiter) //

```

Comment:
I would like to keep like that because it's the natural snippet that you could write with vscode. My idea is that if someone write snippet for vscode, you can just copy/paste the snippet and it works directly.